lib/Sema/TypeCheckType.cpp

```diff
@@ -969,9 +971,18 @@ static Type resolveTypeDecl(TypeDecl *typeDecl, SourceLoc loc,
   auto &diags = ctx.Diags;
   auto lazyResolver = ctx.getLazyResolver();
 
+  Type type = Type();
+
+  // Use only the structural type of an alias decl in structural mode
+  auto aliasDecl = dyn_cast<TypeAliasDecl>(typeDecl);
+  if (resolution.getStage() == TypeResolutionStage::Structural &&
+      aliasDecl && !aliasDecl->getGenericParams() &&
+      !aliasDecl->getUnderlyingTypeLoc().wasValidated()) {
```

Comment:
There's a similar check in getStructuralType() in GenericSignatureBuilder.cpp. Maybe TypeAliasDecl::getStructuralType() should automatically return the checked underlying type if its available.

Reply:
I've applied the suggested change to TypeAliasDecl::getStructuralType. I'll leave this check here because we still want to execute TypeChecker::resolveTypeInContext if the type was previously validated, it provides better diagnostics.